service/timestreamquery/api_client.go

```diff
@@ -69,12 +81,19 @@ type Options struct {
 	// The credentials object to use when signing requests.
 	Credentials aws.CredentialsProvider
 
+	// Allows configuring endpoint discovery
+	EnableEndpointDiscovery aws.EndpointDiscoveryEnableState
+
 	// The endpoint options to be used when attempting to resolve an endpoint.
 	EndpointOptions EndpointResolverOptions
 
 	// The service endpoint resolver.
 	EndpointResolver EndpointResolver
 
+	// Allows configuring an endpoint resolver to use when attempting an endpoint
+	// discovery api request.
+	EndpointResolverUsedForDiscovery EndpointResolver
+
```

Comment:
are these only added for APIs where endpoint discovery is required?   I could seem the being useful where endpoint discovery is optional for same reason as APIs where its required.

Reply:
answered in the comment above.